Allow JVM debug options to be configurable (#18758)
gradle/gradle: subprojects/docs/src/docs/userguide/running-builds/build_environment.adoc

When set to `true`, Gradle will run the build with remote debugging enabled, listening on port 5005. Note that this is the equivalent of adding `-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005` to the JVM command line and will suspend the virtual machine until a debugger is attached.
_Default is `false`._

`org.gradle.debug.port=(port number)`::

> Specifies the port number on which process is listening when debug is enabled.

Reviewer: ```suggestion
Specifies the port number to listen on when debug is enabled.
```

_Default is `5005`._

`org.gradle.debug.server=(true,false)`::

> Then set to `true` and debugging enabled, Gradle will run the build with socked-attach mode of debugger. Otherwise socket-listen is used.

Reviewer: ```suggestion
If set to `true` and debugging is enabled, Gradle will run the build with the socket-attach mode of the debugger. Otherwise, the socket-listen mode is used.
```

_Default is `true`._

`org.gradle.debug.suspend=(true,false)`::

> When set to `true` and debugging enabled, Gradle will suspend the virtual machine until a debugger is attached.

Reviewer: ```suggestion
When set to `true` and debugging is enabled, the JVM running Gradle will suspend until a debugger is attached.
```
